src/Traits/SingleTableInheritanceTrait.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace SamIT\Yii2\Traits;
+
+use SamIT\Yii2\Queries\SingleTableInheritanceQuery;
+
+trait SingleTableInheritanceTrait
+{
+    /**
+     * Returns an array with keys 'map' and 'column'
+     * where 'map' contains a map of type => class
+     * and 'column' contains the column name
+     * [
+     *      'map' => [
+     *          \app\Car::class => 'car'
+     *      ],
+     *      'column' => 'type'
+     * ]
+     * This function should return the same map every time
+     * @return array
+     */
+    protected static function inheritanceConfig() {
+        throw new \Exception('Inheritance config must be implemented');
```

Comment:
oh oops, did not see the static :)
btw: an interface can define a static method and if you implement it you can work around this limitation. Not needed in this case though.